solr/core/src/java/org/apache/solr/handler/loader/JsonLoader.java

```diff
@@ -249,14 +251,27 @@ public void handle(Map<String, Object> record, String path) {
     private SolrInputDocument buildDoc(Map<String, Object> m) {
       SolrInputDocument result = new SolrInputDocument();
       for (Map.Entry<String, Object> e : m.entrySet()) {
-        if (e.getKey() == null) {// special case. JsonRecordReader emits child docs with null key
+        if (entryIsChildDoc(e.getValue())) {// special case. JsonRecordReader emits child docs with null key
```

Comment:
the comment is concerning; we don't want JsonRecordReader to do that anymore?

Reply:
I changed the dictionary implementation to insert child documents into the map under their key, instead of null, so their relation to the parent document can be saved and used when SolrInputDocument is constructed.